### PR TITLE
Switch to api for loading event/participant info

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Removes a direct query and internal function call in favor of the api.

Before
----------------------------------------
BAO & SQL :(

After
----------------------------------------
API :)

Technical Details
----------------------------------------
Renames some of the array keys we were passing around within the webform to match the standard keys from the api.

### CODE REVIEW

File: includes/wf_crm_webform_postprocess.inc [hunk 1]
```diff
@@ -487,12 +486,12 @@ class wf_crm_webform_postprocess extends wf_crm_webform_base {
         if ($event['ended']) {
           form_set_error($eid, t('Sorry, you can no longer register for %event.', array('%event' => $event['title'])));
```
Comment (kainuk) [inline includes/wf_crm_webform_postprocess.inc]: `form_set_error` is removed from Drupal8 - you can get this working by using  `$this->form_state->setErrorByName` (The form_set_error was already there before the port)

File: includes/wf_crm_webform_postprocess.inc [hunk 2]
```diff
@@ -487,12 +486,12 @@ class wf_crm_webform_postprocess extends wf_crm_webform_base {
...
         }
-        elseif ($event['max_participants'] && $event['count'] > $event['remaining']) {
-          if (!empty($event['full'])) {
-            form_set_error($eid, '<em>' . $event['title'] . '</em>: ' . $event['full_message']);
```
Comment (kainuk) [inline includes/wf_crm_webform_postprocess.inc:492]: again `form_set_error`. The drupal 8 alternative does not support html so the emphasis tags   give the following effect. 
Selection_003

File: includes/wf_crm_webform_postprocess.inc [hunk 3]
```diff
@@ -487,12 +486,12 @@ class wf_crm_webform_postprocess extends wf_crm_webform_base {
...
+        elseif (!empty($event['max_participants']) && $event['count'] > $event['available_places']) {
+          if (!empty($event['is_full'])) {
+            form_set_error($eid, '<em>' . $event['title'] . '</em>: ' . $event['event_full_text']);
           }
           else {
-            form_set_error($eid, format_plural($event['remaining'],
```
Comment (kainuk) [inline includes/wf_crm_webform_postprocess.inc:495]: And this is the last `form_set_error`.

**Comment by kainuk** [comment]
Just tested it with the replaced `form_set_error`. Event full works.

**Comment by kainuk** [comment]
Line items is still a bit a mystery for me, but with this PR will be a improvement.

**Comment by colemanw** [comment]
@kainuk so is this ready to merge in your opinion?

**Comment by kainuk** [comment]
Hi @colemanw I created a review, but I forgat to submit it. I have added a number of Drupal 8 changes. I added them as a diff file to this comment.
[PR247diff.txt](https://github.com/colemanw/webform_civicrm/files/3559729/PR247diff.txt)

**Comment by colemanw** [comment]
@kainuk I've manually applied your patch but in the future you can just submit a PR against a PR.

**Comment by KarinG** [comment]
Thank you both!

**Pull request merged**